api/gql_schema.py

```diff
@@ -76,6 +79,43 @@ class Meta:
         fields = "__all__"
 
 
+class CreatePostWithSerializer(SerializerMutation):
+    class Meta:
+        serializer_class = NewPostRequestSerializer
+        model_operations = ["create"]
+        # lookup_field = 'id'
+
+    @classmethod
+    def get_serializer_kwargs(cls, root, info, **input):
+        # Gets messy
+        input["profile"] = info.context.user.profile
+        kwargs = super(CreatePostWithSerializer, cls).get_serializer_kwargs(
+            root, info, **input
+        )
+        return kwargs
+
+
+class CreatePost(graphene.Mutation):
```

Comment:
this seems simple enough and would be really easy to migrate from our current viewsets. 
The only thing is DRF model serializers have better built validation that might need to happen manually here, maybe pre-service calls